### PR TITLE
Prevent automatic page change when using history

### LINKED ISSUE #75774: rustdoc: Cannot go back page when enable "I'm feeling lucky" setting
Enable "Directly go to item in search if there is only one result" option on Settings
image

Search an unique item in search box generated by rustdoc, it will lead you to the doc of that item.
Click the go back button in browser, the page will change to previous search result and go forward 
to the doc of the searched item.

How to reproduce the issue:
* Go to std doc: https://doc.rust-lang.org/nightly/std/index.html
* Search for "PhantomData"
* Click the go back button in browser.

### Meta

`rustc --version --verbose`: 1.47.0-nightly (e15510ca3 2020-08-20)

cc @GuillaumeGomez

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #75774.

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -1576,14 +1576,18 @@ function defocusSearchBar() {
         }
 
         function showResults(results) {
-            if (results.others.length === 1 &&
-                getCurrentValue("rustdoc-go-to-only-result") === "true") {
+            var search = getSearchElement();
+            if (results.others.length === 1
+                && getCurrentValue("rustdoc-go-to-only-result") === "true"
+                && (!search.firstChild || search.firstChild.innerText !== getSearchLoadingText()))
```
Comment (pickfire) [inline src/librustdoc/html/static/main.js:1582]: In what cases will `!search.firstChild` happen?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js:1582]: By default it's empty. So it's not empty when you already performed a search and "cancelled" it (meaning, you pressed ESC or remove all the content from the search input) or if there is the "loading search result" text when you arrive on the page with a search parameter in your URL.

Comment (pickfire) [reply src/librustdoc/html/static/main.js:1582]: Does not look that straightforward to me but I understand after you explain.

**Review by pickfire — APPROVED** [review]
Looks good to me.

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js:1582]: I'll add a comment then.

**Comment by GuillaumeGomez** [comment]
I added an explanation on the condition.

**Review by pickfire — APPROVED** [review]
Good and even clearer now.

**Comment by GuillaumeGomez** [comment]
clang failed? O.o

Well, I guess it's out of the scope of this PR so let's retry...

@bors: retry

**Comment by GuillaumeGomez** [comment]
cc @rust-lang/infra

**Comment by pietroalbini** [comment]
We're already on it. [Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/MSVC.20on.20CI.20has.20broke).

@bors retry

**Comment by GuillaumeGomez** [comment]
Oh great! Thanks for the quick answer. :)

**Pull request merged**